util/chunk/mutrow_test.go

```diff
@@ -134,3 +134,53 @@ func BenchmarkMutRowFromValues(b *testing.B) {
 		MutRowFromValues(values)
 	}
 }
+
+func (s *testChunkSuite) TestMutRowShadowCopyPartialRow(c *check.C) {
+	colTypes := make([]*types.FieldType, 0, 8)
+	colTypes = append(colTypes, &types.FieldType{Tp: mysql.TypeVarString})
+	colTypes = append(colTypes, &types.FieldType{Tp: mysql.TypeVarString})
+	colTypes = append(colTypes, &types.FieldType{Tp: mysql.TypeLonglong})
+	colTypes = append(colTypes, &types.FieldType{Tp: mysql.TypeLonglong})
+	colTypes = append(colTypes, &types.FieldType{Tp: mysql.TypeDatetime})
+
+	mutRow := MutRowFromTypes(colTypes)
+	row := MutRowFromValues("abc", "abcdefg", 123, 456, types.ZeroDatetime).ToRow()
+	mutRow.ShadowCopyPartialRow(0, row)
+	c.Assert(row.GetString(0), check.DeepEquals, mutRow.ToRow().GetString(0))
+	c.Assert(row.GetString(1), check.DeepEquals, mutRow.ToRow().GetString(1))
+	c.Assert(row.GetInt64(2), check.DeepEquals, mutRow.ToRow().GetInt64(2))
```

Comment:
Why we use DeepEquals here?

Reply:
We'd better modify a value of `row`, and then check whether the corresponding value in `mutRow` is also modified.